### PR TITLE
Improve Test Coverage for --take-ownership Flag in Helm Commands

### LINKED ISSUE #13519: Improve test coverage for TakeOwnership option
While reviewing #13439 to add flags for the SDK function added in #12876, I found a bug only through manual testing. I recommended the fix in this comment https://github.com/helm/helm/pull/13439#pullrequestreview-2494390079 and it was added to the PR. However, this means our test coverage for this feature needs to improve. A test needs to be added for `upgrade --install` with `--take-ownership`, as I outlined in this comment https://github.com/helm/helm/pull/13439#pullrequestreview-2494359393.

### PR DESCRIPTION
**Pull request by AnanyaDevGo**
This PR enhances the test coverage for the --take-ownership flag introduced in [#13439](https://github.com/helm/helm/pull/13439). It addresses gaps in testing for the helm upgrade --install --take-ownership functionality, ensuring both success and failure scenarios are adequately covered.

closes #13519 

Key Changes:
Setup Test Environment:

Added a SetupTestEnvironment utility to create a fake Kubernetes namespace for testing.
Ensures namespace cleanup after test execution.
Integration Tests for --take-ownership:

Validates ownership transfer in scenarios involving Helm-managed resources.
Tests edge cases with both successful and failing updates when --take-ownership is toggled.
Enhanced Client Behavior:

Ensures accurate simulation of Kubernetes client operations using FailingKubeClient.
Bug Reproduction & Fix Validation:

Verifies the resolution of a prior issue where helm upgrade --install --take-ownership failed to handle existing unmanaged resources correctly.
Tested locally  and passed all the testcases
Checklist:
- [ ] This PR contains user-facing changes
- [x] This PR contains unit tests
- [ ] This PR has been tested for backward compatibility
Signed-off-by: Ananya E P areduvail@gmail.com

### CODE REVIEW

**Comment by yxxhero** [comment]
@AnanyaDevGo please fix DCO issue by following https://github.com/helm/helm/pull/13610/checks?check_run_id=35295228694.

File: go.mod [hunk 1]
```diff
@@ -24,7 +24,7 @@ require (
 	github.com/lib/pq v1.10.9
 	github.com/mattn/go-shellwords v1.0.12
 	github.com/mitchellh/copystructure v1.2.0
-	github.com/moby/term v0.5.2
+	github.com/moby/term v0.5.1
```
Comment (ivelichkovich) [inline go.mod:27]: Why the downgrade?

File: pkg/kube/fake/fake_test.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+			client := &FailingKubeClient{
+				PrintingKubeClient: PrintingKubeClient{
+					Out: os.Stdout,
+				},
+				UpdateError: tt.updateError,
+			}
+
+			existingList := kube.ResourceList{existingResource}
+			modifiedList := kube.ResourceList{modifiedResource}
+			result, err := client.Update(existingList, modifiedList, tt.takeOwnership)
```
Comment (ivelichkovich) [inline pkg/kube/fake/fake_test.go:107]: Please excuse my ignorance if I'm off base, I'm new to this repo... 

FailingKubeClient.Update() https://github.com/helm/helm/blob/main/pkg/kube/fake/fake.go#L110 just calls PrintingKubeClient.Update() https://github.com/helm/helm/blob/main/pkg/kube/fake/printer.go#L93. This isn't going to actually do anything with --take-ownership as far as I can tell

so I don't think existingList or modifiedList are modified so later when you check the ownerreferences you're really just checking the ownerrefs from your original list that you're passing in

File: pkg/testutils/helm_test.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	if err != nil {
+		t.Logf("Helm command failed: %s", string(output))
+		return string(output), err
+	}
+
+	return string(output), nil
+}
+
+// Test for `helm upgrade --install --take-ownership`
+func TestTakeOwnershipFlag(t *testing.T) {
```
Comment (ivelichkovich) [inline pkg/testutils/helm_test.go:52]: I don't know this is testing helm's --take-ownership flag. This is just creating a pod with helm ownerref and then checking if the pod was created with that ownerref

File: pkg/kube/fake/fake_test.go (hunk 2 above)
Comment (ivelichkovich) [reply pkg/kube/fake/fake_test.go:107]: IIUC takeownership doesn't use ownerrefs but rather the metadata (annotations/labels) for ownership

Comment (ivelichkovich) [reply pkg/kube/fake/fake_test.go:107]: This function shows what that metadata is: https://github.com/helm/helm/blob/main/pkg/action/validate.go#L94
